tests/integration-tests/tests/networking/test_on_demand_capacity_reservation.py

```diff
@@ -0,0 +1,79 @@
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+
+import boto3
+import pytest
+from assertpy import assert_that
+from utils import describe_cluster_instances, retrieve_cfn_resources
+
+
+@pytest.mark.usefixtures("os", "region")
+def test_on_demand_capacity_reservation(
+    region, pcluster_config_reader, placement_group_stack, clusters_factory, odcr_stack
+):
+    """Verify open, targeted and pg odcrs can be created and instances can be launched into them."""
+
+    """This test is only for slurm."""
+
+    resource_groups_client = boto3.client("resource-groups")
+    odcr_resources = retrieve_cfn_resources(odcr_stack.name, region)
+    resource_group_arn = resource_groups_client.get_group(Group=odcr_stack.cfn_resources["integTestsOdcrGroup"])[
+        "Group"
+    ]["GroupArn"]
+
+    cluster_config = pcluster_config_reader(
+        placement_group=placement_group_stack.cfn_resources["PlacementGroup"],
+        open_capacity_reservation_id=odcr_resources["integTestsOpenOdcr"],
+        open_capacity_reservation_arn=resource_group_arn,
+        target_capacity_reservation_id=odcr_resources["integTestsTargetOdcr"],
+        target_capacity_reservation_arn=resource_group_arn,
+        pg_capacity_reservation_id=odcr_resources["integTestsPgOdcr"],
+        pg_capacity_reservation_arn=resource_group_arn,
+    )
+    cluster = clusters_factory(cluster_config)
+
+    _assert_instance_in_capacity_reservation(cluster, region, "open-odcr-id-cr", odcr_resources["integTestsOpenOdcr"])
+    _assert_instance_in_capacity_reservation(cluster, region, "open-odcr-arn-cr", odcr_resources["integTestsOpenOdcr"])
+    _assert_instance_in_capacity_reservation(
+        cluster, region, "open-odcr-id-pg-cr", odcr_resources["integTestsOpenOdcr"]
+    )
+    _assert_instance_in_capacity_reservation(
+        cluster, region, "open-odcr-arn-pg-cr", odcr_resources["integTestsOpenOdcr"]
+    )
+    _assert_instance_in_capacity_reservation(
+        cluster, region, "target-odcr-id-cr", odcr_resources["integTestsTargetOdcr"]
+    )
+    _assert_instance_in_capacity_reservation(
+        cluster, region, "target-odcr-arn-cr", odcr_resources["integTestsTargetOdcr"]
+    )
+    _assert_instance_in_capacity_reservation(
+        cluster, region, "target-odcr-id-pg-cr", odcr_resources["integTestsTargetOdcr"]
+    )
+    _assert_instance_in_capacity_reservation(
+        cluster, region, "target-odcr-arn-pg-cr", odcr_resources["integTestsTargetOdcr"]
+    )
+    _assert_instance_in_capacity_reservation(cluster, region, "pg-odcr-id-cr", odcr_resources["integTestsPgOdcr"])
+    _assert_instance_in_capacity_reservation(cluster, region, "pg-odcr-arn-cr", odcr_resources["integTestsPgOdcr"])
+
+    # need to delete the cluster before deleting placement group
+    cluster.delete()
```

Comment:
This is optional. In the test function definition, you can put `cluster_factory` to the last to make sure it is deleted first
For example:
```
def test_on_demand_capacity_reservation(
    region, pcluster_config_reader, placement_group_stack, odcr_stack, clusters_factory,
):
```

Reply:
updated ty!